src/passes/OptimizeInstructions.cpp

```diff
@@ -1904,6 +1920,39 @@ struct OptimizeInstructions
     }
   }
 
+  // Tries pushing the struct.new down so that it is closer
+  // to a potential struct.set.
+  bool trySwap(ExpressionList& list, Index i, Index j) {
+    if (j == list.size() - 1) {
+      // There is no reason to swap with the last element
+      // of the list as it won't match the pattern.
```

Comment:
Because there is nothing after it? Makes sense, but please add that to the comment.

Nit, please make comments reach column 80, rather than shortening them when you don't have to (unless there is a particular reason for special indentation).